src/librbd/image/OpenRequest.cc

```diff
@@ -277,7 +277,10 @@ Context *OpenRequest<I>::handle_v2_get_immutable_metadata(int *result) {
                << cpp_strerror(*result) << dendl;
     send_close_image(*result);
   } else {
-    send_v2_get_stripe_unit_count();
+    if (m_image_ctx->features & RBD_FEATURE_STRIPINGV2)
```

Comment:
@gmayyyha Wait, did you test this patch? Because the features would be applied in refresh step. so the features is not already set at this place.

Reply:
@trociny @yangdongsheng fixed. I think calling `get_stripe_unit_count` should be before init_layout rather than after refresh

Reply:
Could you explain the reason why we can't get striping information after refresh?

Reply:
Okey, after a look at init_layout() it's true.